isolates/bin/long_running_isolate.dart

```diff
@@ -0,0 +1,88 @@
+// Spawn an isolate, read multiple files, send their contents to the spawned
+// isolate, and wait for the parsed JSON.
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+import 'dart:isolate';
+
+final fileNames = [
+  'json_01.json',
+  'json_02.json',
+  'json_03.json',
+];
+
+// Reads multiple files and parses them on a separate isolate.
+Future<void> main() async {
+  // Read some data.
+  final fileData = await _readFiles();
+
+  // Use that data.
+  await for (final jsonData in _parseJsonStrings(fileData)) {
+    print('Received JSON with ${jsonData.length} keys');
+  }
+}
+
+// Reads multiple files simultaneously using Future.wait().
+Future<List<String>> _readFiles() async {
+  return await Future.wait([
+    ...fileNames.map((f) => File(f)).map((file) => file.readAsString()),
+  ]);
+}
+
+// Sends multiple JSON strings to a separate isolate.
+Stream<Map<String, dynamic>> _parseJsonStrings(
+    List<String> jsonStrings) async* {
+  final p = ReceivePort();
+  await Isolate.spawn(_receiveAndParseJsonService, p.sendPort);
+
+  // Send each JSON string in order, waiting for the previous result before
+  // sending the next JSON string.
+  var i = 0;
+
+  // The port used to send JSON strings to be parsed.
+  // This is the first message sent by the spawned isolate.
+  late final SendPort sendPort;
```

Comment:
I'd personally avoid using `late`, but using `await for` on a heterogenous stream makes it tricky.
This is ... OK.
I'd probably use [`StreamQueue` from `package:async`](https://pub.dev/documentation/async/latest/async/StreamQueue-class.html) instead, it's more readable:
```dart
  var events = StreamQueue<dynamic>(p);
  SendPort sendPort = await events.next();
  for (var jsonString in jsonStrings) {
    sendPort.send(jsonString);
    Map<String, dynamic> message = await events.next();
    yield message;
  }
  sendPort.send(null);
  await events.cancel();
```

Reply:
That's much better, added in 2cbb16d